colcon_output/event_handler/event_log.py

```diff
@@ -45,7 +46,16 @@ def __call__(self, event):  # noqa: D102
             '[%f] (%s) %s: %s\n' % (
                 self._get_relative_time(), context,
                 data.__class__.__name__, members))
-        self._file_handle.flush()
+        try:
+            self._file_handle.flush()
+        except OSError as e:
+            if e.errno == errno.ENOSPC:
+                # for known error code suppress the stacktrace
+                # and only show the exception message
+                raise RuntimeError(
+                    str(e) +
+                    ' [{self._file_handle.name}]'.format_map(locals()))
```

Comment:
It might be useful to retain the entire stacktrace in the new exception for debugging. Users shouldn't see it since you're suppressing the stacktrace anyway.

```suggestion
                raise RuntimeError(
                    str(e) +
                    ' [{self._file_handle.name}]'.format_map(locals())) from e
```

Reply:
Not sure I follow. Since the calling code doesn't show the stacktrace for `RuntimeError`s I guess this doesn't make a difference. So what is the reason to add the `from e`?

Reply:
Honestly, just because it is good practice to preserve as much stack information as possible when rethrowing exceptions in general.
As I noted and you also pointed out, the user won't see it anyway, so I figured it might be a good pattern to follow.
You seem to think it isn't worth it, but I don't see a downside. In any case, I already approved, so feel free to keep it as-is if you don't believe that practice applies here.

Reply:
If I read the `from e` I would expect "oh, this wants to make sure the original stacktrace is preseved". But the way it is used that is not the case. So in this particular case I think I will go without it.